### PR TITLE
SSH tunnel connection: PostgreSQL and Redshift

### PR DESCRIPTION
**Pull request by ishachopra0401**
## What do these changes do?

> Describe the changes you are proposing in this PR.

## Type of documentation

- [ ] New documentation
- [x] Documentation update
- [ ] Hotfix

### CODE REVIEW

File: docs/destinations/warehouse-destinations/redshift.mdx [hunk 1]
```diff
@@ -157,6 +157,27 @@ To configure Redshift as a destination in RudderStack, follow these steps:
...
+This feature is available only for the <a href="https://www.rudderstack.com/pricing/">Growth and Enterprise plan</a> users.
+</div>
+
+RudderStack lets you connect to your Redshift database securely over an SSH connection. Enter the following details to do the same:
+
+- **SSH Connection**: Toggle on this boolean setting to use the SSH connection while connecting to your Redshift database.
+- **SSH Host**: Enter the IP address of your bastion host.
+- **SSH Port**: Enter the port for the above host.
+- **SSH User**: Enter the user name of your Redshift database.
+- **SSH Public Key**: Copy the public key provided in this field and add it to the `authorizated_keys` file on your bastion host. Rudderstack will use the private key corresponding to this public key to establish the connection successully.
```
Comment (abhimanyubabbar) [inline docs/destinations/warehouse-destinations/redshift.mdx:173]: Should we also show snapshot of how the screen looks when we capture the ssh credentials ?

Comment (ishachopra0401) [reply docs/destinations/warehouse-destinations/redshift.mdx:175]: Added an image without the credentials though. We will need to blur those even if we capture. LMK what do you think.

File: docs/destinations/warehouse-destinations/postgresql.mdx [hunk 2]
```diff
@@ -51,6 +51,29 @@ To configure PostgreSQL as a destination in RudderStack, follow these steps:
...
 - **Port**: Enter the port number associated with your PostgreSQL instance.
 - **Namespace**: Enter the schema name where RudderStack will create all the tables. If you don't specify any namespace, RudderStack will set this to the source name, by default.
+
+### SSH connection
+
+<div class="infoBlock">
+This feature is available only for the <a href="https://www.rudderstack.com/pricing/">Growth and Enterprise plan</a> users.
+</div>
+
+SSH tunneling is a method of transporting data over an encrypted SSH connection. You can use it to add encryption to your legacy applications as well as achieve compliance such as HIPAA, PCI-DSS, and other standards without having the need to modify the existing applications.
```
Comment (abhimanyubabbar) [inline docs/destinations/warehouse-destinations/postgresql.mdx:61]: Can we change it to `transferring` ?

Comment (abhimanyubabbar) [reply docs/destinations/warehouse-destinations/postgresql.mdx:61]: transporting -> transferring

File: docs/destinations/warehouse-destinations/redshift.mdx [hunk 3]
```diff
@@ -157,6 +157,29 @@ To configure Redshift as a destination in RudderStack, follow these steps:
...
+SSH tunneling is a method of transporting data over an encrypted SSH connection. You can use it to add encryption to your legacy applications as well as achieve compliance such as HIPAA, PCI-DSS, and other standards without having the need to modify the existing applications.
+
+RudderStack lets you connect to your Redshift database securely over an SSH connection. Enter the following details to do the same:
+
+<img src="../../assets/warehouse-destinations/ssh-connection.png" alt="Redshift edit configuration" />
+
+- **SSH Connection**: Toggle on this boolean setting to use the SSH connection while connecting to your Redshift database.
+- **SSH Host**: Enter the IP address of your bastion host.
+- **SSH Port**: Enter the port for the above host.
+- **SSH User**: Enter the user name of your Redshift database.
```
Comment (abhimanyubabbar) [inline docs/destinations/warehouse-destinations/redshift.mdx:176]: `Enter the username used to access the bastion host`

File: docs/destinations/warehouse-destinations/postgresql.mdx [hunk 4]
```diff
@@ -51,6 +51,29 @@ To configure PostgreSQL as a destination in RudderStack, follow these steps:
...
+
+RudderStack lets you connect to your PostgreSQL database securely over an SSH connection. Enter the following details to do the same:
+
+<img src="../../assets/warehouse-destinations/ssh-connection.png" alt="Redshift edit configuration" />
+
+- **SSH Connection**: Toggle on this boolean setting to use the SSH connection while connecting to your PostgreSQL database.
+- **SSH Host**: Enter the IP address of your bastion host.
+- **SSH Port**: Enter the port for the above host.
+- **SSH User**: Enter the user name of your PostgreSQL database.
```
Comment (abhimanyubabbar) [inline docs/destinations/warehouse-destinations/postgresql.mdx:70]: `Enter the username used to access the bastion host`

**Review by abhimanyubabbar — COMMENTED** [review]
Some changes suggested. Let me know what you think ?

**Comment by ishachopra0401** [comment]
@abhimanyubabbar Incorporated all.

File: docs/destinations/warehouse-destinations/postgresql.mdx (hunk 2 above)
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/postgresql.mdx]:
```suggestion
SSH tunneling is a method of transporting data over an encrypted SSH connection. You can use it to add encryption to your legacy applications and achieve compliance like HIPAA, PCI-DSS, and other standards, without having to modify the existing applications.
```

File: docs/destinations/warehouse-destinations/postgresql.mdx [hunk 5]
```diff
@@ -51,6 +51,29 @@ To configure PostgreSQL as a destination in RudderStack, follow these steps:
...
+
+### SSH connection
+
+<div class="infoBlock">
+This feature is available only for the <a href="https://www.rudderstack.com/pricing/">Growth and Enterprise plan</a> users.
+</div>
+
+SSH tunneling is a method of transporting data over an encrypted SSH connection. You can use it to add encryption to your legacy applications as well as achieve compliance such as HIPAA, PCI-DSS, and other standards without having the need to modify the existing applications.
+
+RudderStack lets you connect to your PostgreSQL database securely over an SSH connection. Enter the following details to do the same:
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/postgresql.mdx:63]:
```suggestion
RudderStack lets you connect to your PostgreSQL database securely over an SSH connection by configuring these settings:
```

File: docs/destinations/warehouse-destinations/postgresql.mdx [hunk 6]
```diff
@@ -51,6 +51,29 @@ To configure PostgreSQL as a destination in RudderStack, follow these steps:
...
+
+<img src="../../assets/warehouse-destinations/ssh-connection.png" alt="Redshift edit configuration" />
+
+- **SSH Connection**: Toggle on this boolean setting to use the SSH connection while connecting to your PostgreSQL database.
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/postgresql.mdx:67]:
```suggestion
- **SSH Connection**: Enable this setting to use the SSH connection while connecting to your PostgreSQL database.
```

File: docs/destinations/warehouse-destinations/postgresql.mdx [hunk 7]
```diff
@@ -51,6 +51,29 @@ To configure PostgreSQL as a destination in RudderStack, follow these steps:
...
+- **SSH Host**: Enter the IP address of your bastion host.
+- **SSH Port**: Enter the port for the above host.
+- **SSH User**: Enter the user name of your PostgreSQL database.
+- **SSH Public Key**: Copy the public key provided in this field and add it to the `authorizated_keys` file on your bastion host. Rudderstack will use the private key corresponding to this public key to establish the connection successully.
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/postgresql.mdx:71]:
```suggestion
- **SSH Public Key**: Copy the public key provided in this field and add it to the `authorizated_keys` file on your bastion host. RudderStack uses the private key corresponding to this public key to establish the connection successfully.
```

Comment (ameypv-rudder) [reply docs/destinations/warehouse-destinations/postgresql.mdx:71]: Should it be `authorized_keys`?

File: docs/destinations/warehouse-destinations/postgresql.mdx [hunk 8]
```diff
@@ -51,6 +51,29 @@ To configure PostgreSQL as a destination in RudderStack, follow these steps:
...
+
+You can also enable the SSH connection for an existing PostgreSQL destination by navigating to the destination's **Configuration** tab, selecting **Edit configuration** and enabling the **SSH connection** setting.
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/postgresql.mdx:73]:
```suggestion

#### Enabling SSH connection for existing destinations

To enable the SSH connection for an existing PostgreSQL destination, navigate to the destination's **Configuration** tab, select **Edit configuration** and enable the **SSH connection** setting.
```

Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/postgresql.mdx:73]:
```suggestion
You can also enable the SSH connection for an existing PostgreSQL destination by navigating to the destination's **Configuration** tab, selecting **Edit configuration** and enabling the **SSH connection** setting.

#### Enabling SSH connection for existing destinations

```

File: docs/destinations/warehouse-destinations/postgresql.mdx [hunk 9]
```diff
@@ -51,6 +51,29 @@ To configure PostgreSQL as a destination in RudderStack, follow these steps:
...
+RudderStack lets you connect to your PostgreSQL database securely over an SSH connection. Enter the following details to do the same:
+
+<img src="../../assets/warehouse-destinations/ssh-connection.png" alt="Redshift edit configuration" />
+
+- **SSH Connection**: Toggle on this boolean setting to use the SSH connection while connecting to your PostgreSQL database.
+- **SSH Host**: Enter the IP address of your bastion host.
+- **SSH Port**: Enter the port for the above host.
+- **SSH User**: Enter the user name of your PostgreSQL database.
+- **SSH Public Key**: Copy the public key provided in this field and add it to the `authorizated_keys` file on your bastion host. Rudderstack will use the private key corresponding to this public key to establish the connection successully.
+
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/postgresql.mdx:72]:
```suggestion

#### Enabling SSH connection for existing destinations

```

Comment (ameypv-rudder) [reply docs/destinations/warehouse-destinations/postgresql.mdx:72]: IMO, making this a subsection will be helpful. WDYT?

File: docs/destinations/warehouse-destinations/redshift.mdx [hunk 10]
```diff
@@ -157,6 +157,29 @@ To configure Redshift as a destination in RudderStack, follow these steps:
...
 - **Password**: The password for the above user.
 - **Namespace**: Enter the schema name where RudderStack will create all the tables. If you don't specify any namespace, RudderStack will set this to the source name, by default.
+
+### SSH connection
+
+<div class="infoBlock">
+This feature is available only for the <a href="https://www.rudderstack.com/pricing/">Growth and Enterprise plan</a> users.
+</div>
+
+SSH tunneling is a method of transporting data over an encrypted SSH connection. You can use it to add encryption to your legacy applications as well as achieve compliance such as HIPAA, PCI-DSS, and other standards without having the need to modify the existing applications.
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/redshift.mdx]:
```suggestion
SSH tunneling is a method of transporting data over an encrypted SSH connection. You can use it to add encryption to your legacy applications and achieve compliance like HIPAA, PCI-DSS, and other standards, without having to modify the existing applications.
```

File: docs/destinations/warehouse-destinations/redshift.mdx [hunk 11]
```diff
@@ -157,6 +157,29 @@ To configure Redshift as a destination in RudderStack, follow these steps:
...
+
+RudderStack lets you connect to your Redshift database securely over an SSH connection. Enter the following details to do the same:
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/redshift.mdx:169]:
```suggestion
RudderStack lets you connect to your Redshift database securely over an SSH connection by configuring these settings:
```

File: docs/destinations/warehouse-destinations/redshift.mdx [hunk 12]
```diff
@@ -157,6 +157,29 @@ To configure Redshift as a destination in RudderStack, follow these steps:
...
+
+<img src="../../assets/warehouse-destinations/ssh-connection.png" alt="Redshift edit configuration" />
+
+- **SSH Connection**: Toggle on this boolean setting to use the SSH connection while connecting to your Redshift database.
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/redshift.mdx:173]:
```suggestion
- **SSH Connection**: Enable this setting to use the SSH connection while connecting to your Redshift database.
```

File: docs/destinations/warehouse-destinations/redshift.mdx (hunk 1 above)
Comment (ameypv-rudder) [reply docs/destinations/warehouse-destinations/redshift.mdx]: Same question as above - should it be `authorized_keys`? just verifying if this isn't a typo.

File: docs/destinations/warehouse-destinations/redshift.mdx [hunk 13]
```diff
@@ -157,6 +157,29 @@ To configure Redshift as a destination in RudderStack, follow these steps:
...
+- **SSH Host**: Enter the IP address of your bastion host.
+- **SSH Port**: Enter the port for the above host.
+- **SSH User**: Enter the user name of your Redshift database.
+- **SSH Public Key**: Copy the public key provided in this field and add it to the `authorizated_keys` file on your bastion host. Rudderstack will use the private key corresponding to this public key to establish the connection successully.
+
+You can also enable the SSH connection for an existing Amazon Redshift destination by navigating to the destination's **Configuration** tab, selecting **Edit configuration** and enabling the **SSH connection** setting.
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/redshift.mdx:179]: @ishachopra0401 same change as in the Postgres doc if we agree on the structure.

Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/redshift.mdx:179]:
```suggestion
To enable the SSH connection for an existing Redshift destination, navigate to the destination's **Configuration** tab, select **Edit configuration** and enable the **SSH connection** setting.
```

**Review by ameypv-rudder — APPROVED** [review]

File: docs/destinations/warehouse-destinations/postgresql.mdx [hunk 14]
```diff
@@ -51,6 +51,29 @@ To configure PostgreSQL as a destination in RudderStack, follow these steps:
...
 - **Port**: Enter the port number associated with your PostgreSQL instance.
 - **Namespace**: Enter the schema name where RudderStack will create all the tables. If you don't specify any namespace, RudderStack will set this to the source name, by default.
+
+### SSH connection
+
+<div class="infoBlock">
+This feature is available only for the <a href="https://www.rudderstack.com/pricing/">Growth and Enterprise plan</a> users.
+</div>
+
+SSH tunneling is a method of transferring data over an encrypted SSH connection. You can use it to add encryption to your legacy applications as well as achieve compliance such as HIPAA, PCI-DSS, and other standards without having the need to modify the existing applications.
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/postgresql.mdx:61]:
```suggestion
SSH tunneling is a method of transferring data over an encrypted SSH connection. You can use it to add encryption to your legacy applications and achieve compliance with the regulations like HIPAA, PCI-DSS, etc., without having to modify the existing applications.
```

File: docs/destinations/warehouse-destinations/redshift.mdx [hunk 15]
```diff
@@ -157,6 +157,29 @@ To configure Redshift as a destination in RudderStack, follow these steps:
...
 - **Password**: The password for the above user.
 - **Namespace**: Enter the schema name where RudderStack will create all the tables. If you don't specify any namespace, RudderStack will set this to the source name, by default.
+
+### SSH connection
+
+<div class="infoBlock">
+This feature is available only for the <a href="https://www.rudderstack.com/pricing/">Growth and Enterprise plan</a> users.
+</div>
+
+SSH tunneling is a method of transferring data over an encrypted SSH connection. You can use it to add encryption to your legacy applications as well as achieve compliance such as HIPAA, PCI-DSS, and other standards without having the need to modify the existing applications.
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/redshift.mdx:167]:
```suggestion
SSH tunneling is a method of transferring data over an encrypted SSH connection. You can use it to add encryption to your legacy applications and achieve compliance with regulations like HIPAA, PCI-DSS, etc., without having to modify the existing applications.
```

File: docs/destinations/warehouse-destinations/postgresql.mdx (hunk 9 above)
Comment (ishachopra0401) [reply docs/destinations/warehouse-destinations/postgresql.mdx:72]: This is common to all the source and destination settings, like not an important piece of info even, and just a one-liner. I think its fine if we dont include a header.

**Comment by ishachopra0401** [comment]
@BadriV1 I have added the introduction on why someone should use SSh tunneling. Can you please add some details on the different ways to configure Postgres?

**Pull request merged**